### PR TITLE
Support xi-core's UpdateOp::update + some refactoring

### PR DESCRIPTION
**Pull request by c4eater**
The main objective of this PR is to catch up with UpdateOp::update implementation in xi-core (xi-editor/xi-editor#1302). The upstream xrl (v0.0.8) has 2 problems preventing it from handling UpdateOp::update correctly: it does not allow empty "styles" in the "line" json (despite the frontend protocol spec), and it does not handle the "ln" (which is needed in the update op just as it is needed in the copy op if the line numbers are shifted).

Also OperationType::Copy_ is renamed to Copy for consistency, which implies update of public API and incrementation of the minor version number.

Also the LineCache is slightly refactored as described in the correspondent commit.

### CODE REVIEW

**Comment by c4eater** [comment]
@Cogitri please also review this one?

**Comment by Cogitri** [comment]
lgtm I suppose, tests still pass and the benchmark speed hasn't changed. Let's see if @little-dude or @bytebuddha have suggestions for this.

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ license-file = "LICENSE-MIT"
 name = "xrl"
 readme = "README.md"
 repository = "https://github.com/xi-frontend/xrl"
-version = "0.0.8"
+version = "0.1.0"
```
Comment (c4eater) [inline Cargo.toml:14]: On second thought, I'd change this to 0.0.9. As xrl is below v1.0 I guess it's ok to break API between build versions.

**Comment by Cogitri** [comment]
Apparently I'm added as owner on crates.io too, so I could release the new version I think

**Comment by Cogitri** [comment]
Updated the crates.io version :heavy_check_mark:

**Pull request merged**